### PR TITLE
Bug 1767163: fix app.kubernetes.io/name to use just IS name

### PR DESCRIPTION
**Pull request by soltysh**
Fixing problem introduced in https://github.com/openshift/oc/pull/101

/assign @adambkaplan @DhritiShikhar

### CODE REVIEW

**Comment by DhritiShikhar** [comment]
Tested this PR locally. 

`app.kubernetes.io/name` label is **NOT** set.

```
➜  oc git:(change-label) ./oc new-app https://github.com/openshift/ruby-hello-world.git -i=ruby
--> Found image c237e73 (2 weeks old) in image stream "openshift/ruby" under tag "2.5" for "ruby"
                                                           
    Ruby 2.5                  
    -------- 
    Ruby 2.5 available as container is a base platform for building and running various Ruby 2.5 applications and frameworks. Ruby is the interpreted scripting language for quick and easy object-oriented programming. It has many features 
to process text files and to do system management tasks (as in Perl). It is simple, straight-forward, and extensible.
                                                           
    Tags: builder, ruby, ruby25, rh-ruby25           
                                                                                                                       
    * The source repository appears to match: ruby         
    * A source build using source code from https://github.com/openshift/ruby-hello-world.git will be created
      * The resulting image will be pushed to image stream tag "ruby-hello-world:latest"
      * Use 'oc start-build' to trigger a new build
    * This image will be deployed in deployment config "ruby-hello-world"
    * Port 8080/tcp will be load balanced by service "ruby-hello-world"
      * Other containers can access this service through the hostname "ruby-hello-world"

--> Creating resources ...
    imagestream.image.openshift.io "ruby-hello-world" created
    buildconfig.build.openshift.io "ruby-hello-world" created
    deploymentconfig.apps.openshift.io "ruby-hello-world" created
    service "ruby-hello-world" created
--> Success
    Build scheduled, use 'oc logs -f bc/ruby-hello-world' to track its progress.
    Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
     'oc expose svc/ruby-hello-world' 
    Run 'oc status' to view your app.
➜  oc git:(change-label) oc get dc               
NAME               REVISION   DESIRED   CURRENT   TRIGGERED BY
ruby-hello-world   0          1         0         config,image(ruby-hello-world:latest)
➜  oc git:(change-label) oc describe dc ruby-hello-world
Name:           ruby-hello-world
Namespace:      project-1
Created:        17 seconds ago
Labels:         app=ruby-hello-world
                app.kubernetes.io/component=ruby-hello-world
                app.kubernetes.io/instance=ruby-hello-world
Annotations:    openshift.io/generated-by=OpenShiftNewApp
Latest Version: Not deployed
Selector:       deploymentconfig=ruby-hello-world
Replicas:       1
Triggers:       Config, Image(ruby-hello-world@latest, auto=true)
Strategy:       Rolling
Template:
Pod Template:
  Labels:       deploymentconfig=ruby-hello-world
  Annotations:  openshift.io/generated-by=OpenShiftNewApp
  Containers:
   ruby-hello-world:
    Image:              ruby-hello-world:latest
    Port:               8080/TCP
    Host Port:          0/TCP
    Environment:        <none>
    Mounts:             <none>
  Volumes:              <none>

Latest Deployment:      <none>

Events: <none>
```

File: pkg/cli/newapp/newapp.go [hunk 1]
```diff
@@ -375,8 +375,8 @@ func (o *AppOptions) RunNewApp() error {
 			"app.kubernetes.io/component": result.Name,
 		}
 
-		for _, name := range o.Config.ImageStreams {
-			if name != "" {
+		for _, is := range o.Config.ImageStreams {
+			if name, _, ok := imageutil.SplitImageStreamTag(is); ok && len(name) > 0 {
```
Comment (adambkaplan) [inline pkg/cli/newapp/newapp.go]: @soltysh I don't think we need to check `ok` here. `latest` is an implicit tag name if not provided - we just need to strip out the `:` if it is present.

**Comment by adambkaplan** [comment]
PR to verify that new-app works with an imagestream: https://github.com/openshift/origin/pull/24096

File: pkg/cli/newapp/newapp.go (hunk 1 above)
Comment (soltysh) [reply pkg/cli/newapp/newapp.go]: @adambkaplan yeah, that's the problem I just fixed in the last update, ptal

**Pull request merged**